lib/galaxy/datatypes/binary.py

```diff
@@ -810,6 +812,83 @@ def display_peek(self, dataset):
+    MetadataElement(name="format", default=None, desc="format", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="type", default=None, desc="table type", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="generated_by", default=None, desc="generated by", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="creation_date", default=None, desc="creation date", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="nnz", default=None, desc="nnz: The number of non-zero elements in the table", readonly=True, visible=True, no_value=None)
+    MetadataElement(name="shape", default=None, desc="shape: The number of rows and columns in the dataset", readonly=True, visible=True, no_value=None)
+
+    file_ext = "biom2"
+    edam_format = "format_3746"
+
+    def sniff(self, filename):
+        """
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'biom2_sparse_otu_table_hdf5.biom' )
+        >>> Biom2().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'test.mz5' )
+        >>> Biom2().sniff( fname )
+        False
+        >>> fname = get_test_fname( 'wiggle.wig' )
+        >>> Biom2().sniff( fname )
+        False
+        """
+        if super(Biom2, self).sniff(filename):
+            try:
+                f = h5py.File(filename)
+                attributes = list(dict(f.attrs.items()))
+                required_fields = ['id', 'format-url', 'type', 'generated-by', 'creation-date', 'nnz', 'shape']
+                return set(required_fields).issubset(attributes)
+            except Exception:
+                return False
+        return False
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        super(Biom2, self).set_meta(dataset, overwrite=overwrite, **kwd)
+        try:
+            f = h5py.File(dataset.file_name)
+            attributes = dict(f.attrs.items())
+
+            dataset.metadata.id = attributes['id']
+            dataset.metadata.format_url = attributes['format-url']
+            if 'format-version' in attributes:  # biom 2.1
+                dataset.metadata.format_version = '.'.join(map(str, list(attributes['format-version'])))
+            elif 'format' in attributes:  # biom 2.0
+                dataset.metadata.format = attributes['format']
+            dataset.metadata.type = attributes['type']
+            dataset.metadata.shape = str(list(attributes['shape']))
+            dataset.metadata.generated_by = attributes['generated-by']
+            dataset.metadata.creation_date = attributes['creation-date']
+            dataset.metadata.nnz = str(attributes['nnz'])
+
+        except Exception as e:
+            log.warning('%s, set_meta Exception: %s', self, e)
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Biom2 (HDF5) file"
```

Comment:
Maybe we can add here a little bit metadata as well.

Reply:
good idea